sktime/performance_metrics/base/_base.py

```diff
@@ -15,14 +16,53 @@ class BaseMetric(BaseObject):
     Extends sktime BaseObject.
     """
 
+    # todo: 0.13.0, remove the func/name args here and to all the metrics
     def __init__(
         self,
-        func,
+        func=None,
         name=None,
     ):
+
+        if func is not None or name is not None:
+            warn(
+                "func and name parameters of BaseMetric are deprecated from 0.12.0"
+                "and will be removed in 0.13.0"
+            )
```

Comment:
Shouldn't this have the deprecation warning class in it? 
(Same for all other deprecation warnings in the PR)

```python 
from warnings import warn, DeprecationWarning

warn(
    "func and name parameters of BaseMetric are deprecated from 0.12.0"
    "and will be removed in 0.13.0", 
    DeprecationWarning
)
```

Reply:
yes, it should!

Reply:
I always forget that...